File: 프로그래머스/lv1/장정윤/기사 단원의 무기.py
import math
def solution(number, limit, power):
    answer = 0
    
    # 1 2 2 3 2 => 합산 10
    # 1 2 2 3 2 4 2 4 3 4 => 3보다 큰 수는 power 수로 대치 10 11 = 21
    cnt_list = []
    for i in range(1, number+1):
        cnt_num = 0
        # for j in range(1, i+1):
        #     if i % j == 0:
        #         cnt_num += 1
        # i 의 제곱근
        for j in range(1, int(math.sqrt(i)) + 1):
            if i % j == 0:
                cnt_num += 1
                if j != i // j:  # 서로 다른 약수인 경우
                    cnt_num += 1
        cnt_list.append(cnt_num)
        
    for k in range(len(cnt_list)):
        if cnt_list[k] > limit:
            cnt_list[k] = power
    
    return sum(cnt_list)

File: 프로그래머스/lv1/김원래/기사단원의 무기.cpp
#include <string>
#include <vector>

using namespace std;

int numberOfDivision[100001];

int solution(int number, int limit, int power) {
    int answer = 0;
    
    //number 전체를 도는 반복문
    for(int num = 1; num <= number ; ++num)
    {

        //num의 약수를 세는 반목문
        for(int modNum = 1 ; modNum * modNum <= num ; ++modNum)
        {
            if(num % modNum == 0)
            {
                if(modNum * modNum == num)
                {
                    numberOfDivision[num]++;
                }
                else
                {
                    numberOfDivision[num] += 2;
                }
            }
        }
    }

    
    for(int num = 1 ; num <= number ; ++num)
    {
        //printf("%d, ",numberOfDivision[num]);
        
        if(numberOfDivision[num] > limit) answer += power;
        else answer += numberOfDivision[num];
    }
    
    
    return answer;
}